Assets/MixedRealityToolkit/_Core/Interfaces/IMixedRealitySceneObject.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.MixedReality.Toolkit.Core.Interfaces
+{
+    /// <summary>
+    /// Generic interface for all optional Mixed Reality systems, components, or features that can be added to the <see cref="Definitions.MixedRealityComponentConfiguration"/>
+    /// </summary>
+    public interface IMixedRealitySceneObject 
```

Comment:
Should we add the `TrackedObjectType` to this? 

Might make it easier to determine the type of object that's registered.

Reply:
I don't think so, as the TrackedObject attachment is unique to the solvers atm.
Plus, (eventually) this could just contain regular scene objects that don't care whether they are tracked.